src/test/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/ApplicationTest.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.companieshouse.chipsrestinterfacesconsumer;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.common.ApplicationLogger;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.service.LoopingMessageProcessor;
+
+import java.util.concurrent.CompletableFuture;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class ApplicationTest {
+
+    @Mock
+    private ApplicationLogger logger;
+
+    @Mock
+    private LoopingMessageProcessor loopingMainMessageConsumer;
+
+    @Mock
+    private LoopingMessageProcessor loopingRetryMessageConsumer;
+
+    @InjectMocks
+    private Application application;
+
+    @Test
+    void runErrorProcessingTrueTest() {
+        ReflectionTestUtils.setField(application, "isErrorConsumer", true);
+        application.run();
+
+        verify(loopingMainMessageConsumer, times(0)).loopReadAndProcess();
+        verify(loopingRetryMessageConsumer, times(0)).loopReadAndProcess();
```

Comment:
put TODOs in the tests (matching the TODO in Application) to verify error consumer is / is not called